### PR TITLE
chore: use carthage 0.40.0 - no ticket

### PR DESCRIPTION
**Pull request by caldrian**
### Changes

This PR updates the carthage.sh script to use version 0.40.0.

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: scripts/carthage.sh [hunk 1]
```diff
@@ -20,19 +20,20 @@ set -Eeuo pipefail
 #
 
 REPO_ROOT=$(git rev-parse --show-toplevel)
-CARTHAGE_URL="https://github.com/Carthage/Carthage/releases/download/0.39.1/Carthage.pkg"
+CARTHAGE_URL="https://github.com/Carthage/Carthage/releases/download/0.40.0/Carthage.pkg"
 CARTHAGE_DIR="$REPO_ROOT/Carthage"
-CARTHAGE="$CARTHAGE_DIR/carthage"
+CARTHAGE="$CARTHAGE_DIR/carthage-0.40.0"
```
Comment (netbe) [inline scripts/carthage.sh:25]: why this change?

Comment (caldrian) [reply scripts/carthage.sh:25]: So that it actually updates carthage, otherwise the check below
```
 [[ ! -f "$CARTHAGE" ]]; then
```
would not evaluate to true even if the present version is below 0.40.0.

We can revert it if you prefer.

Comment (netbe) [reply scripts/carthage.sh:25]: no that's fine, I just wanted to understand

**Review by netbe — APPROVED** [review]

**Review by agisilaos — APPROVED** [review]
